prime-router/src/test/kotlin/fhirengine/engine/FhirTranslatorTests.kt

```diff
@@ -142,6 +183,130 @@ class FhirTranslatorTests {
+        val actionHistory = mockk<ActionHistory>()
+        val actionLogger = mockk<ActionLogger>()
+        val engine = makeSendOriginalFhirEngine()
+        mockkObject(BlobAccess.Companion)
+
+        val parentReportId = UUID.randomUUID()
+        val childReportId = UUID.randomUUID()
+        val rootItemLineage =
+            ItemLineage(
+                9000000125356546,
+                null,
+                0,
+                parentReportId,
+                0,
+                "trackingId1",
+                null,
+                OffsetDateTime.now(),
+                null
+            )
+        val childItemLineage =
+            ItemLineage(
+                9000000125356546,
+                parentReportId,
+                0,
+                childReportId,
+                0,
+                "trackingId2",
+                null,
+                OffsetDateTime.now(),
+                null
+            )
+        val report = ReportFile().setReportId(parentReportId).setBodyUrl("testingurl").setSchemaTopic(Topic.ELR_ELIMS)
+        val reportContent = "reportContent"
+        val message =
+            spyk(
+                FhirTranslateQueueMessage(
+                    UUID.randomUUID(),
+                    BLOB_URL,
+                    "test",
+                    BLOB_SUB_FOLDER,
+                    topic = Topic.ELR_ELIMS,
+                    originalSenderOrganization.receivers[0].fullName
+                )
+            )
+        val bodyFormat = Report.Format.FHIR
+        val bodyUrl = BODY_URL
+
+        // need a different mock for the one that is instantiated within the method
+        mockkConstructor(WorkflowEngine::class)
+        mockkConstructor(DatabaseAccess::class)
+
+        val dbAccessMock = mockk<DatabaseAccess>()
+        every {
+            anyConstructed<WorkflowEngine>().db
+        }.returns(dbAccessMock)
+        every {
+            dbAccessMock.fetchItemLineagesForReport(any(), any(), any())
+        }.returnsMany(listOf(childItemLineage), listOf(rootItemLineage))
+        every { downloadBlobAsByteArray(any()) }.returns(reportContent.toByteArray())
+        every { actionLogger.hasErrors() } returns false
+        every { message.downloadContent() }
+            .returns(File(VALID_DATA_URL).readText())
+        every { BlobAccess.Companion.uploadBlob(any(), any()) } returns "test"
+        every { dbAccessMock.fetchReportFile(any(), any(), any()) }.returns(report)
+        every { accessSpy.insertTask(any(), bodyFormat.toString(), bodyUrl, any()) }.returns(Unit)
+        every { actionHistory.trackCreatedReport(any(), any(), blobInfo = any()) }.returns(Unit)
+        every { actionHistory.trackExistingInputReport(any()) }.returns(Unit)
+        every { actionHistory.trackActionReceiverInfo(any(), any()) }
+            .returns(Unit)
+
+        accessSpy.transact { txn ->
+            engine.run(message, actionLogger, actionHistory, txn)
+        }
+
+        verify(exactly = 1) {
```

Comment:
I'm not sure what the value of this test is? Seems like we may as well just leave it to the smokeTests at this point.